### ISSUE TITLE
opencv build failure

### ISSUE THREAD

**Issue by DougRogers** [microsoft/vcpkg#42598]
### Operating system

Windows 11

### Compiler

MSVC

### Steps to reproduce the behavior

```Shell
vcpkg install opencv[tbb,ipp,contrib,eigen,dnn,ipp,nonfree,lapack]:x64-windows --recurse
```

### Failure logs

-- Configuring x64-windows
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:127 (message):
    Command failed: "C:/Program Files/Microsoft Visual Studio/2022/Community/Common7/IDE/CommonExtensions/Microsoft/CMake/Ninja/ninja.exe" -v
    Working Directory: C:/vcpkg/buildtrees/opencv4/x64-windows-rel/vcpkg-parallel-configure
    Error code: 1
    See logs for more information:
      C:\vcpkg\buildtrees\opencv4\config-x64-windows-dbg-CMakeCache.txt.log
      C:\vcpkg\buildtrees\opencv4\config-x64-windows-rel-CMakeCache.txt.log
      C:\vcpkg\buildtrees\opencv4\config-x64-windows-dbg-CMakeConfigureLog.yaml.log
      C:\vcpkg\buildtrees\opencv4\config-x64-windows-rel-CMakeConfigureLog.yaml.log
      C:\vcpkg\buildtrees\opencv4\config-x64-windows-out.log

Call Stack (most recent call first):
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_configure.cmake:269 (vcpkg_execute_required_process)
  ports/opencv4/portfile.cmake:332 (vcpkg_cmake_configure)
  scripts/ports.cmake:196 (include)

error: building opencv4:x64-windows failed with: BUILD_FAILED
See https://learn.microsoft.com/vcpkg/troubleshoot/build-failures?WT.mc_id=vcpkg_inproduct_cli for more information.
Elapsed time to handle opencv4:x64-windows: 46 s
Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
Then check for known issues at:
  https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+opencv4
You can submit a new issue at:
  https://github.com/microsoft/vcpkg/issues/new?title=[opencv4]+Build+error+on+x64-windows&body=Copy+issue+body+from+C%3A%2Fvcpkg%2Finstalled%2Fvcpkg%2Fissue_body.md

[config-x64-windows-out.log](https://github.com/user-attachments/files/18055185/config-x64-windows-out.log)

**Comment by dg0yt** [comment]
Do you depend on feature `lapack`?
~~~
-- Found OpenBLAS libraries: C:/vcpkg/installed/x64-windows/lib/openblas.lib
-- Found OpenBLAS include: C:/vcpkg/installed/x64-windows/include/openblas
-- LAPACK(OpenBLAS): LAPACK_LIBRARIES: C:/vcpkg/installed/x64-windows/lib/openblas.lib
CMake Warning at cmake/OpenCVFindLAPACK.cmake:52 (message):
  LAPACK(OpenBLAS): CBLAS/LAPACK headers are not found in
  'C:/vcpkg/installed/x64-windows/include/openblas'
Call Stack (most recent call first):
  cmake/OpenCVFindLAPACK.cmake:188 (ocv_lapack_check)
  CMakeLists.txt:831 (include)
...
CMake Warning at cmake/OpenCVUtils.cmake:785 (message):
  Option WITH_LAPACK is enabled but corresponding dependency have not been
  found: "HAVE_LAPACK" is FALSE
Call Stack (most recent call first):
  CMakeLists.txt:1974 (ocv_verify_config)
~~~

It is probably not a regression from the recent openblas update. The default BLAS+LAPACK installation for x64-windows is openblas + lapack-reference. This did not and does not provide LAPACKE (`lapacke.h`) as far as I can tell.
(Addressing LAPACKE was the background for applying cleanup to openblas. But this still needs work.)

**Comment by DougRogers** [comment]
I removed the lapack requirement and it built successfully.   I have build using the lapack option for months in a script, with no build issues before.

**Comment by dg0yt** [comment]
What changed now? openblas, opencv4, other?

**Comment by jonolds** [comment]
exact same here. I'm on windows 11. I set up a clean manifest to see if I could figure out how you guys work around unrelated problems I'm having. My manifest is:

`{
  "builtin-baseline": "20a72ce99b12dd0ebfea5d39f32681bd68b19d03",
  "dependencies": [
	{
      "name": "opencv4",
      "features": [
        "ade",
        "contrib",
        "cuda",
        "cudnn",
        "dnn-cuda",
        "dshow",
        "ffmpeg",
        "freetype",
        "gstreamer",
        "halide",
        "ipp",
        "lapack",
        "msmf",
        "nonfree",
        "opencl",
        "openexr",
        "opengl",
        "openjpeg",
        "openmp",
        "openvino",
        "qt",
        "rgbd",
        "sfm",
        "tbb",
        "thread",
        "vtk",
        "vulkan",
        "win32ui",
        "world"
      ]
    }
  ]
}`

**Comment by LilyWangLL** [comment]
This issue could be reproduced on the latest vcpkg.
cc @cenit Could you please take a look at this issue?

**Comment by cenit** [comment]
lapack was not tested in ci. Added into #42371, which will fix it

**Comment by dg0yt** [comment]
LAPACK needs work with regard to opencv. When OpenCV finds OpenBLAS, it expects it to also bring LAPACKE, but this is not the current state of the packages.

**Comment by cenit** [comment]
same for lapack as an alternative package to openblas, when it finds it, it expects it containing lapacke.

Since Eigen is used as an alternative in most of the cases, I'd say it's a very reasonable alternative while waiting for full lapack support

**Comment by jobmart** [comment]
Step 1: Update Vcpkg
Ensure your Vcpkg is up to date. Run the following commands:
`cd C:\vcpkg
git pull
vcpkg update
`
Step 2: Install Dependencies
Some dependencies may be missing. Install them as follows:

Ensure Visual Studio 2022 is installed with the following components:

Desktop development with C++ workload.
The following individual components:
MSVC v143 - VS 2022 C++ x64/x86 build tools.
CMake tools for Windows.
Windows 10 or 11 SDK.
Install additional tools:

Install Python (for OpenCV’s build system): [Download Python](https://www.python.org/downloads/).
Install Ninja (if not already installed): vcpkg install ninja.
Step 3: Clear Previous Builds
Clear any residual files that might cause conflicts:
`cd C:\vcpkg
vcpkg remove --outdated
vcpkg remove opencv4
rd /s /q buildtrees
`
Step 4: Verify Features
Double-check the features you're attempting to install. Some combinations may cause conflicts. For instance, using ipp and lapack together might require additional configurations. Try the following minimal command first:
`vcpkg install opencv[core,dnn,eigen]:x64-windows --recurse
`
Once this works, incrementally add features.
Step 5: Examine Logs
Review the logs mentioned in the error message for detailed errors:

Open the following files for error details:
C:\vcpkg\buildtrees\opencv4\config-x64-windows-dbg-CMakeConfigureLog.yaml.log
C:\vcpkg\buildtrees\opencv4\config-x64-windows-out.log
Look for missing files, incorrect paths, or unresolved dependencies.
Step 6: Run CMake Manually
Sometimes, diagnosing the issue requires running the CMake command directly:

Go to the OpenCV build directory:
bash
Copy code
`cd C:\vcpkg\buildtrees\opencv4\x64-windows-rel
`
Run the command manually
`cmake -G "Ninja" -DCMAKE_TOOLCHAIN_FILE=C:/vcpkg/scripts/buildsystems/vcpkg.cmake ..
`
Address any errors that arise.
Step 7: Increase Verbosity
Enable verbose output to understand the issue better:
`vcpkg install opencv[core,dnn,eigen]:x64-windows --recurse --x-use-aria2 --debug
`
Step 8: Community Resources
If the problem persists:

Check for related issues on the [Vcpkg GitHub Issues page](https://github.com/microsoft/vcpkg/issues?q=opencv).
Submit a new issue following the instructions provided in the error log.

**Comment by DougRogers** [comment]
Still fails with

vcpkg install opencv[tbb,ipp,contrib,eigen,dnn,ipp,nonfree,lapack]:x64-windows --recurse

**Comment by dg0yt** [comment]
> Still fails with
> 
> vcpkg install opencv[tbb,ipp,contrib,eigen,dnn,ipp,nonfree,lapack]:x64-windows --recurse

Yeah, no lapacke changes in #42371, but https://github.com/microsoft/vcpkg/issues/42598#issuecomment-2539917697.

**Comment by cenit** [comment]
> Still fails with
> 
> vcpkg install opencv[tbb,ipp,contrib,eigen,dnn,ipp,nonfree,lapack]:x64-windows --recurse

technically you should not even be able to find the lapack feature anymore. Did I miss anything?

**Comment by DougRogers** [comment]
The command:
`vcpkg search opencv`

lists it as an option:
`opencv4[lapack]                           LAPACK support for opencv`

**Comment by BillyONeal** [comment]
> The command: `vcpkg search opencv`
> 
> lists it as an option: `opencv4[lapack] LAPACK support for opencv`

You need to `git pull`

**Comment by DougRogers** [comment]
Removed after pull
